gateway/src/model.rs

```diff
@@ -586,6 +588,63 @@ impl<'de> Deserialize<'de> for CredentialLocation {
     }
```

Comment:
^^ up to 561. Now that we're changing the semantics of CredentialLocation so that it is being globally converted into credentials, we should distinguish between the cases where the thing in the environment is literally the key or the path to a file. Path is correct in that it is a hardcoded. But for compatibility with existing deployments we'll also want PathFromEnv(String) that reads the path from environment, then reads the file at that path during the conversion. Can you add that? We'll need it to fix GCP integrations.